soweego/importer/imdb_dump_extractor.py

```diff
@@ -0,0 +1,390 @@
+        ]
+
+        db_manager = DBManager()
+        LOGGER.info("Connected to database: %s", db_manager.get_engine().url)
+
+        db_manager.drop(tables)
+        db_manager.create(tables)
+
+        LOGGER.info("SQL tables dropped and re-created: %s",
+                    [table.__tablename__ for table in tables])
+
+        LOGGER.info("Starting to import movies from imdb dump")
+
+        # Here we open the movie dump file, and add everything to the DB
+        with gzip.open(movies_file_path, "rt") as mdump:
+            reader = csv.DictReader(mdump, delimiter="\t")
+
+            # count number of rows for TQDM (so we can display how)
+            # much is missing to complete the process. Then go back
+            # to the start of the file with `.seek(0)`
+            n_rows = sum(1 for line in mdump)
+            mdump.seek(0)
+
+            LOGGER.debug("Movies dump has %d entries", n_rows)
+
+            session = db_manager.new_session()
+
+            for movie_info in tqdm(reader, total=n_rows):
+                self._normalize_null(movie_info)
+
+                # create the movie SQLAlchemy entity and populate it
+                movie_entity = imdb_entity.ImdbMovieEntity()
+                movie_entity.catalog_id = movie_info.get("tconst")
+                movie_entity.title_type = movie_info.get("titleType")
+                movie_entity.primary_title = movie_info.get("primaryTitle")
+                movie_entity.original_title = movie_info.get("originalTitle")
+                movie_entity.is_adult = True if movie_info.get(
+                    "isAdult") == "1" else False
+                movie_entity.start_year = movie_info.get("startYear")
+                movie_entity.end_year = movie_info.get("endYear")
+                movie_entity.runtime_minutes = movie_info.get("runtimeMinutes")
+
+                if movie_info.get("genres"):  # if movie has a genre specified
+                    movie_entity.genres = movie_info.get("genres").replace(",", " ")
+
+                session.add(movie_entity)
+
+                # every `_sqlalchemy_commit_every` loops we commit the session to
+                # the DB. This is more efficient than commiting every loop, and
+                # is not so hard on the memory requirements as would be
+                # adding everything to session and commiting once the for loop
+                # is done
+                if self.n_movies % self._sqlalchemy_commit_every == 0:
+                    session.commit()
+
+                self.n_movies += 1
+
+            # commit remaining entities
+            session.commit()
+
+        end = datetime.datetime.now()
+        LOGGER.info("Movie import completed in %s. "
+                    "Total movies imported: %d",
+                    end - start, self.n_movies)
+
+        LOGGER.info("Starting import persons from IMDB dump")
+
+        # read person dump and add everything to DB
+        with gzip.open(person_file_path, "rt") as pdump:
```

Comment:
It would be optimal to refactor code that gets duplicated from the previous dump file import, if applicable.

Reply:
Good point. I've extracted the similar parts of the file import to a separate method in 3953fcddea2a0744a4f2f1331d1e0478e0ddc1e3